lib_ux_nbgl/ux.h

```diff
@@ -0,0 +1,101 @@
+
+/*******************************************************************************
+ *   Ledger Nano S - Secure firmware
+ *   (c) 2022 Ledger
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ********************************************************************************/
+
+#pragma once
+
+#include <string.h>
+
+#include "os_math.h"
+#include "os_ux.h"
+#include "os_task.h"
+#include "nbgl_lns.h"
+typedef struct bagl_element_e bagl_element_t;
+
+// callback returns NULL when element must not be redrawn (with a changing color or what so ever)
+typedef const bagl_element_t *(*bagl_element_callback_t)(const bagl_element_t *element);
+
+// a graphic element is an element with defined text and actions depending on user touches
+struct bagl_element_e {
+    bagl_component_t component;
+    const char      *text;
+};
+
+void io_seproxyhal_display_icon(const bagl_component_t    *icon_component,
+                                const bagl_icon_details_t *icon_details);
+
+void io_seproxyhal_display_default(const bagl_element_t *element);
+
+// Kept for retro-compatibility
+typedef struct {
+} ux_state_t;
```

Comment:
No way to tackle this :thinking: ?

Reply:
Thsi si what is done on the "classical" NBGL already: https://github.com/LedgerHQ/ledger-secure-sdk/blob/master/lib_ux_nbgl/ux_nbgl.h#L41
This is to avoid having to add some `#ifdef HAVE_BAGL` in the apps.
I'm fine with not defining it and therefore enforcing a few more changes in the apps.